File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void temp(TreeNode* node, unordered_map<int, vector<int>>& mp, int depth){
        if(!node) return;
        mp[depth].push_back(node->val);
        temp(node->left, mp, depth+1);
        temp(node->right, mp, depth+1);
    }

    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        if(!root) return res;
        unordered_map<int, vector<int>> mp;
        temp(root, mp, 0);
        for(int i=0;i<mp.size();i++){
            res.push_back(mp[i]);
        }
        return res;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
import 'dart:collection';
class Solution {
  List<List<int>> levelOrder(TreeNode? root) {
    if(root == null) return [];
    List<List<int>> res = [];
    Queue<TreeNode> q = Queue();
    q.add(root);
    while(q.isNotEmpty){
        int size = q.length;
        List<int> temp = [];
        for(int i=0;i<size;i++){
            TreeNode cur = q.removeFirst();
            temp.add(cur.val);
            if(cur.left != null) q.add(cur.left!);
            if(cur.right != null) q.add(cur.right!);
        }
        res.add(List.of(temp));
    }
    return res;
  }
}